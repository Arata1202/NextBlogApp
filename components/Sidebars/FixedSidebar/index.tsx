'use client';

import React from 'react';
import Search from '../Elements/Search';
import Profile from '../Elements/Profile';
import Category from '../Elements/Category';
import News from '../Elements/News';
import Tag from '../Elements/Tag';
import Archive from '../Elements/Archive';
import Popular from '../Elements/Popular';
import Recent from '../Elements/Recent';
import { Article } from '@/libs/microcms';
import Display from '../../Adsense/Display';

type Props = {
  articles?: Article[];
};

export default function FixedSidebar({ articles }: Props) {
  return (
    <div className="lg:col-span-1 lg:w-full lg:h-full">
      <div className="sidebar">
        <Search />
        <Profile />
        <News />
        <Category />
        <Tag />
        <div className="FirstAd mt-5">
          <Display slot="8452341403" />
        </div>
        <Archive />
        <Popular />
        <Recent articles={articles} />
      </div>
    </div>
  );
}
