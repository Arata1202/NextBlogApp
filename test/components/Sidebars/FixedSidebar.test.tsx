import { render } from '@testing-library/react';
import FixedSidebar from '@/components/Sidebars/FixedSidebar';

jest.mock('@/components/Sidebars/Elements/Search');
jest.mock('@/components/Sidebars/Elements/Profile');
jest.mock('@/components/Sidebars/Elements/Category');
jest.mock('@/components/Sidebars/Elements/News');
jest.mock('@/components/Sidebars/Elements/Tag');
jest.mock('@/components/Sidebars/Elements/Archive');
jest.mock('@/components/Sidebars/Elements/Popular');
jest.mock('@/components/Sidebars/Elements/Recent');
jest.mock('@/components/Adsense/Display');

describe('FixedSidebar', () => {
  test('スナップショット（FixedSidebar）', () => {
    const { asFragment } = render(<FixedSidebar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
