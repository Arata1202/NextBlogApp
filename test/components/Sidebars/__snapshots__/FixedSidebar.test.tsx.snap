// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedSidebar スナップショット（FixedSidebar） 1`] = `
<DocumentFragment>
  <div
    class="lg:col-span-1 lg:w-full lg:h-full"
  >
    <div
      class="sidebar"
    >
      <div
        class="FirstAd mt-5"
      />
    </div>
  </div>
</DocumentFragment>
`;
